include/sharpSAT/primitive_types.h

```diff
@@ -8,20 +8,189 @@
 #ifndef SHARP_SAT_PRIMITIVE_TYPES_H_
 #define SHARP_SAT_PRIMITIVE_TYPES_H_
 
+#include <ostream>
```

Comment:
I would prefer if this was `<iosfwd>` and the `operator<<`s were just declared, with implementation in their own TU. Generally, the stream headers are rather costly for compilation times, and we do not need the `operator<<` to be inlined, as writing out debug information is not really performance critical.

Reply:
Done.